src/cockpit/packages.py

```diff
@@ -43,6 +44,29 @@ def directory_items(path):
     return sorted(path.iterdir(), key=lambda item: item.name)
 
 
+def parse_accept_language(accept_language: str) -> List[Tuple[str, str]]:
+    """Parse the Accept-Language header
+
+    Returns an ordered list of (language, weight) pairs.
+
+    https://tools.ietf.org/html/rfc7231#section-5.3.5
+    """
+
+    accept_languages = accept_language.split(',')
+    locales = []
+    for index, language in enumerate(accept_languages):
+        if ';' in language:
+            locale, weight = language.split(';')
```

Comment:
According to the spec, the weight defaults to 1 if it's not present.  Look under "[5.3.1](https://www.rfc-editor.org/rfc/rfc7231#section-5.3.1).  Quality Values"

Something like this seems sensible:

```python
locale, _, weight = accept_language.partition('?q=')
weight = float(weight or 1)
```

Reply:
Right that's the else case.